game/scoring.cpp

```diff
@@ -0,0 +1,129 @@
+#include "yahtzee.h"
+#include <algorithm>
+// Util functions
+int sum(int die[5]) {
+    int sum = 0;
+    for (int i = 0; i < 5; ++i)
+    {
+        sum += die[i];
+    }
+    return sum;
+}
+
+//Funtions for scoring yahtzee hand
+
+// Out is an array initialized to zero in which to hold the count
+int sortDie(int die[5], int out[6]) {
+    for (int i = 0; i < 5; ++i)
+    {
+        out[die[i]-1] += 1;
+    }
+    return 0;
+}
+
+/*
+    scores x points for each die showing x pips
+*/
+int scoreNumber(int die[5], int x) {
```

Comment:
Use a vector. Here you aren't changing the vector, so you will want to pass by constant reference. Signature should look like: `int scoreNumber(const std::vector<int>& die)`